src/regression_tests/regression_test.test.ts

```diff
@@ -0,0 +1,137 @@
+let userTwo: RegressionTestUser;
+let rtConfig: RegressionTestConfig;
+
+beforeAll(() => {
+  fetchMock.disableMocks();
+  rtConfig = getEnvVariables();
+  expect(rtConfig).toBeDefined();
+});
+
+afterAll(async () => {
+  if (userOne) {
+    await expect(removeRegisteredPaymail(userOne.paymail, rtConfig.clientOneURL, ADMIN_XPRIV)).resolves.not.toThrow();
+  }
+
+  if (userTwo) {
+    await expect(removeRegisteredPaymail(userTwo.paymail, rtConfig.clientTwoURL, ADMIN_XPRIV)).resolves.not.toThrow();
+  }
+});
+
+describe('TestRegression', () => {
+  describe('Initialize Shared Configurations', () => {
+    test('Should get sharedConfig for instance one', async () => {
+      paymailDomainInstanceOne = await getPaymailDomain(ADMIN_XPRIV, rtConfig.clientOneURL);
+      expect(paymailDomainInstanceOne).not.toBe('');
+    });
+
+    test('Should get sharedConfig for instance two', async () => {
+      paymailDomainInstanceTwo = await getPaymailDomain(ADMIN_XPRIV, rtConfig.clientTwoURL);
+      expect(paymailDomainInstanceTwo).not.toBe('');
+    });
+  });
+
+  describe('Create Users', () => {
+    test('Should create user for instance one', async () => {
+      const userName = 'instanceOneUser1';
+      userOne = await createUser(userName, paymailDomainInstanceOne, rtConfig.clientOneURL, ADMIN_XPRIV);
+    });
+
+    test('Should create user for instance two', async () => {
+      const userName = 'instanceTwoUser1';
+      userTwo = await createUser(userName, paymailDomainInstanceTwo, rtConfig.clientTwoURL, ADMIN_XPRIV);
+    });
+  });
+
+  describe('Perform Transactions', () => {
+    test(
+      'Send money to instance 1',
+      async () => {
+        const amountToSend = 3;
+
+        const transaction = await sendFunds(
+          rtConfig.clientTwoURL,
+          rtConfig.clientTwoLeaderXPriv,
+          userOne.paymail,
+          amountToSend,
+        );
+        expect(transaction.output_value).toBeLessThanOrEqual(-1);
+
+        const balance = await getBalance(rtConfig.clientOneURL, userOne.xpriv);
+        expect(balance).toBeGreaterThanOrEqual(1);
+
+        const transactions = await getTransactions(rtConfig.clientOneURL, userOne.xpriv);
+        expect(transactions.length).toBeGreaterThanOrEqual(1);
+      },
+      TEST_TIMEOUT_MS,
+    );
+
+    test(
+      'Send money to instance 2',
+      async () => {
+        const transaction = await sendFunds(
+          rtConfig.clientOneURL,
+          rtConfig.clientOneLeaderXPriv,
+          userTwo.paymail,
+          MINIMAL_FUNDS_PER_TRANSACTION,
+        );
+        expect(transaction.output_value).toBeLessThanOrEqual(-1);
+
+        const balance = await getBalance(rtConfig.clientTwoURL, userTwo.xpriv);
+        expect(balance).toBeGreaterThanOrEqual(1);
+
+        const transactions = await getTransactions(rtConfig.clientTwoURL, userTwo.xpriv);
+        expect(transactions.length).toBeGreaterThanOrEqual(1);
+      },
+      TEST_TIMEOUT_MS,
+    );
```

Comment:
These two blocks of code look very similar. 
Maybe a good point would be to make a function like that, and use it twice:

```typescript
async function sendAndVerifyFunds(sourceURL, sourcePrivKey, targetPaymail, targetURL, targetXpriv, minExpectedBalance) {
  const transaction = await sendFunds(sourceURL, sourcePrivKey, targetPaymail, MINIMAL_FUNDS_PER_TRANSACTION);
  expect(transaction.output_value).toBeLessThanOrEqual(-1);

  const balance = await getBalance(targetURL, targetXpriv);
  expect(balance).toBeGreaterThanOrEqual(minExpectedBalance);

  const transactions = await getTransactions(targetURL, targetXpriv);
  expect(transactions.length).toBeGreaterThanOrEqual(1);
}
```
> Don't copy my code without doublecheck! Because I'm not sure if it works. It's just a suggesion 🙈

Reply:
Hmm you're right let me test it

Reply:
tested, extracted it works